### PR TITLE
[BEAM-3884] Use Impulse primitive for Read.

### PR DESCRIPTION
**Pull request by robertwb**
------------------------

Follow this checklist to help us incorporate your contribution quickly and easily:

 - [ ] Format the pull request title like `[BEAM-XXX] Fixes bug in ApproximateQuantiles`, where you replace `BEAM-XXX` with the appropriate JIRA issue, if applicable. This will automatically link the pull request to the issue.
 - [ ] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

It will help us expedite review of your Pull Request if you tag someone (e.g. `@username`) to look at it.

### CODE REVIEW

**Comment by robertwb** [comment]
R: @angoenka

File: sdks/python/apache_beam/transforms/core.py [hunk 1]
```diff
@@ -1894,3 +1895,26 @@ def _create_source(serialized_values, coder):
...
+        if not isinstance(pbegin, pvalue.PBegin):
+          raise TypeError(
+            'Input to Impulse transform must be a PBegin but found %s' % pbegin)
+        return pvalue.PCollection(pbegin.pipeline)
+
+    def get_windowing(self, inputs):
+        return Windowing(GlobalWindows())
+
+    def infer_output_type(self, unused_input_type):
+        return type(None)
```
Comment (angoenka) [inline sdks/python/apache_beam/transforms/core.py]: In java Impulse is of type ByteArray Should we keep it the same here.
https://github.com/apache/beam/blob/master/sdks/java/core/src/main/java/org/apache/beam/sdk/transforms/Impulse.java

Comment (lukecwik) [reply sdks/python/apache_beam/transforms/core.py]: yes, we should.

**Comment by robertwb** [comment]
Currently only the Flink runner understands the impulse transform, so I can't really write tests for it.

File: sdks/python/apache_beam/transforms/core.py (hunk 1 above)
Comment (robertwb) [reply sdks/python/apache_beam/transforms/core.py]: Why? I'd have thought Void would be the obvious choice here. (Changed to bytes for consistency.)

Comment (lukecwik) [reply sdks/python/apache_beam/transforms/core.py]: you can change it to void but need to follow up so that Dataflow honors this.

**Comment by robertwb** [comment]
Run Python Dataflow ValidatesRunner

**Comment by robertwb** [comment]
Run Python Dataflow ValidatesRunner

**Comment by lukecwik** [comment]
I believe this is causing the python postcommit to fail with:
======================================================================
ERROR: test_wordcount_fnapi_it (apache_beam.examples.wordcount_it_test.WordCountIT)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/jenkins/jenkins-slave/workspace/beam_PostCommit_Python_Verify/src/sdks/python/apache_beam/examples/wordcount_it_test.py", line 79, in test_wordcount_fnapi_it
    on_success_matcher=PipelineStateMatcher()))
  File "/home/jenkins/jenkins-slave/workspace/beam_PostCommit_Python_Verify/src/sdks/python/apache_beam/examples/wordcount_fnapi.py", line 128, in run
    result = p.run()
  File "/home/jenkins/jenkins-slave/workspace/beam_PostCommit_Python_Verify/src/sdks/python/apache_beam/pipeline.py", line 395, in run
    self.to_runner_api(), self.runner, self._options).run(False)
  File "/home/jenkins/jenkins-slave/workspace/beam_PostCommit_Python_Verify/src/sdks/python/apache_beam/pipeline.py", line 408, in run
    return self.runner.run_pipeline(self)
  File "/home/jenkins/jenkins-slave/workspace/beam_PostCommit_Python_Verify/src/sdks/python/apache_beam/runners/dataflow/test_dataflow_runner.py", line 48, in run_pipeline
    self.result = super(TestDataflowRunner, self).run_pipeline(pipeline)
  File "/home/jenkins/jenkins-slave/workspace/beam_PostCommit_Python_Verify/src/sdks/python/apache_beam/runners/dataflow/dataflow_runner.py", line 350, in run_pipeline
    super(DataflowRunner, self).run_pipeline(pipeline)
  File "/home/jenkins/jenkins-slave/workspace/beam_PostCommit_Python_Verify/src/sdks/python/apache_beam/runners/runner.py", line 170, in run_pipeline
    pipeline.visit(RunVisitor(self))
  File "/home/jenkins/jenkins-slave/workspace/beam_PostCommit_Python_Verify/src/sdks/python/apache_beam/pipeline.py", line 436, in visit
    self._root_transform().visit(visitor, self, visited)
  File "/home/jenkins/jenkins-slave/workspace/beam_PostCommit_Python_Verify/src/sdks/python/apache_beam/pipeline.py", line 791, in visit
    part.visit(visitor, pipeline, visited)
  File "/home/jenkins/jenkins-slave/workspace/beam_PostCommit_Python_Verify/src/sdks/python/apache_beam/pipeline.py", line 791, in visit
    part.visit(visitor, pipeline, visited)
  File "/home/jenkins/jenkins-slave/workspace/beam_PostCommit_Python_Verify/src/sdks/python/apache_beam/pipeline.py", line 791, in visit
    part.visit(visitor, pipeline, visited)
  File "/home/jenkins/jenkins-slave/workspace/beam_PostCommit_Python_Verify/src/sdks/python/apache_beam/pipeline.py", line 794, in visit
    visitor.visit_transform(self)
  File "/home/jenkins/jenkins-slave/workspace/beam_PostCommit_Python_Verify/src/sdks/python/apache_beam/runners/runner.py", line 165, in visit_transform
    self.runner.run_transform(transform_node)
  File "/home/jenkins/jenkins-slave/workspace/beam_PostCommit_Python_Verify/src/sdks/python/apache_beam/runners/runner.py", line 208, in run_transform
    return m(transform_node)
  File "/home/jenkins/jenkins-slave/workspace/beam_PostCommit_Python_Verify/src/sdks/python/apache_beam/runners/dataflow/dataflow_runner.py", line 586, in run_ParDo
    input_step = self._cache.get_pvalue(transform_node.inputs[0])
  File "/home/jenkins/jenkins-slave/workspace/beam_PostCommit_Python_Verify/src/sdks/python/apache_beam/runners/runner.py", line 286, in get_pvalue
    return self._cache[self.key(pvalue)]
KeyError: (u'read/Read/Impulse', None)

**Comment by robertwb** [comment]
Looking into it. Surprised this was not run as part of "Python Dataflow ValidatesRunner."

**Comment by lukecwik** [comment]
Yes, I also forgot about the postcommit that runs wordcount and got bit by this on a different PR.

**Pull request merged**